src/main/java/com/aws/greengrass/ipc/GetClientDeviceAuthTokenOperationHandler.java

```diff
@@ -0,0 +1,130 @@
+import software.amazon.awssdk.aws.greengrass.model.CredentialDocument;
+import software.amazon.awssdk.aws.greengrass.model.GetClientDeviceAuthTokenRequest;
+import software.amazon.awssdk.aws.greengrass.model.GetClientDeviceAuthTokenResponse;
+import software.amazon.awssdk.aws.greengrass.model.InvalidArgumentsError;
+import software.amazon.awssdk.aws.greengrass.model.InvalidCredentialError;
+import software.amazon.awssdk.aws.greengrass.model.MQTTCredential;
+import software.amazon.awssdk.aws.greengrass.model.ServiceError;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
+import software.amazon.awssdk.eventstreamrpc.model.EventStreamJsonMessage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.aws.greengrass.ipc.common.ExceptionUtil.translateExceptions;
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.GET_CLIENT_DEVICE_AUTH_TOKEN;
+
+public class GetClientDeviceAuthTokenOperationHandler
+        extends GeneratedAbstractGetClientDeviceAuthTokenOperationHandler {
+    private static final Logger logger = LogManager.getLogger(GetClientDeviceAuthTokenOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final String MQTT_CREDENTIAL_TYPE = "mqtt";
+    private static final String NO_DEVICE_CREDENTIAL_ERROR = "Invalid client device credentials";
+    private final String serviceName;
+    private final AuthorizationHandler authorizationHandler;
+    private final SessionManager sessionManager;
+    private final Map<String, String> credentialMap = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param context              operation continuation handler
+     * @param sessionManager       session manager
+     * @param authorizationHandler authorization handler
+     */
+    public GetClientDeviceAuthTokenOperationHandler(
+            OperationContinuationHandlerContext context,
+            SessionManager sessionManager,
+            AuthorizationHandler authorizationHandler
+    ) {
+
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.sessionManager = sessionManager;
+        this.authorizationHandler = authorizationHandler;
+    }
+
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    @Override
+    public GetClientDeviceAuthTokenResponse handleRequest(GetClientDeviceAuthTokenRequest request) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForClientDevAuthToken();
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
```

Comment:
Same thing here. Log the cause but return a boiler plate message

Reply:
The message is already boilerplate defined in the nucleus. All our ipc services do this the same way. It should not be changed.

Reply:
Ah, right, this is core IPC stuff. As long as that error message is controlled in Nucleus (which it looks like it is) then that's ok